doc/concepts/foreign.rst

```diff
@@ -0,0 +1,210 @@
+     (foreign_stubs (language cxx) (names src3) (flags -O2))
+     (foreign_archives arch1 some/dir/arch2))
+
+Here, in addition to :ref:`foreign-stubs`, we also specify foreign archives
+``arch1`` and ``arch2``, where the latter is stored in a subdirectory
+``some/dir``.
+
+You can build a foreign archive manually, e.g., using a custom ``rule`` as
+described in :ref:`foreign-sandboxing`, or ask Dune to build it via the
+``foreign_library`` stanza:
+
+.. code:: dune
+
+    (foreign_library
+     (archive_name arch1)
+     (language c)
+     (names src4 src5)
+     (include_dir headers))
+
+This asks Dune to compile C source files ``src4`` and ``src5`` with
+headers tracked in the ``headers`` directory and put the resulting
+object files into an archive ``arch1``, whose full name is typically
+``libarch1.a`` for static linking and ``dllarch1.so`` for dynamic
+linking.
+
+The ``foreign_library`` stanza supports all :ref:`foreign-stubs` fields plus
+the ``archive_name`` field, which specifies the archive's name. You can refer
+to the same archive name from multiple OCaml libraries and executables, so a
+foreign archive is a bit like a foreign library, hence the name of the stanza.
+
+Foreign archives are particularly useful when embedding a library written in
+a foreign language and/or built with another build system. See
+:ref:`foreign-sandboxing` for more details.
+
+
+.. _extra-objects:
+
+Extra Objects
+-------------
+
+It's possible to specify native object files to be packaged with OCaml
+libraries or linked into OCaml executables. Do this by using the
+``extra_objects`` field of the ``library`` or ``executable`` stanzas.
+For example:
+
+.. code:: dune
+
+    (executable
+     (public_name main)
+     (extra_objects foo bar))
+
+    (rule
+     (targets foo.o bar.o)
+     (deps foo.c bar.c)
+     (action (run ocamlopt %{deps})))
+
+This example builds an executable which is linked against a pair of native
+object files, ``foo.o`` and ``bar.o``. The ``extra_objects`` field takes a
+list of object names, which correspond to the object file names with their path
+and extension omitted.
+
+In this example, the sources corresponding to the objects (``foo.c`` and
+``bar.c``)  are assumed to be present in the same directory as the OCaml source
+code, and a custom ``rule`` is used to compile the C source code into object
+files using ``ocamlopt``. This is not necessary; one can instead compile foreign
+object files manually and place them next to the OCaml source code.
+
+.. _flags-flow:
+
+Flags
+-----
+
+Depending on the :ref:`always-add-cflags` option, the base `:standard` set of
+flags for C will contain only ``ocamlc_cflags`` or both ``ocamlc_cflags`` and
+``ocamlc_cppflags``.
+
+There are multiple levels where one can declare custom flags (using the
+:doc:`ordered-set-language`), and each level inherits the flags of the previous
+one in its `:standard` set:
+
+- In the global `env` definition of a `dune-workspace` file
```

Comment:
```suggestion
- In the global ``env`` definition of a ``dune-workspace`` file
```
Should these be in double `` for monospace in .rst? If so, several following lines will need to be changed too.

Reply:
In .rst, monospace are double backticks and italics are single backticks, yes.
To be even more precise, filenames can be set using the following semantic markup:

```rst
:file:`dune-workspace`
```

(with our configuration this produces an output identical to double backticks)

Reply:
That’s what I thought about the double backticks, so those should be in monospace, right?